lib/options.nix

```diff
@@ -99,6 +103,33 @@ rec {
     type = lib.types.bool;
   };
 
+  /* Creaties an Option attribute set for an option that specifies the
+     package a module should use.
+
+     The argument default is an attribute set path in pkgs.
+  */
+  mkPackageOption =
+    { name ? last default, pkgs, default ? [ name ], example ? null }:
```

Comment:
one of `name` or `default` should lose their default, otherwise forgetting about both will give you infinite recursion. `name` seems to be a good candidate since many packages referred to by these options have length-1 attrpaths.